x-pack/plugins/maps/public/components/widget_overlay/layer_control/layer_toc/view.js

```diff
@@ -5,61 +5,76 @@
  */
 
 import React from 'react';
+import {
+  EuiDragDropContext,
+  EuiDroppable,
+  EuiDraggable,
+} from '@elastic/eui';
 import { TOCEntry } from './toc_entry';
-import $ from 'jquery';
 
 export class LayerTOC extends React.Component {
 
-  constructor(props) {
-    super(props);
-    this._domContainer = null;
-  }
-
-  componentDidMount() {
-    if (!this.props.isReadOnly) {
-      this._attachSortHandler();
+  _onDragEnd = ({ source, destination }) => {
+    // Dragging item out of EuiDroppable results in destination of null
+    if (!destination) {
+      return;
     }
-  }
 
-  _attachSortHandler() {
-    const tocEntries = this._domContainer;
-    let length;
-    $(tocEntries).sortable({
-      start: (evt, { item }) => {
-        length = tocEntries.children.length;
-        $(this).attr('data-previndex', length - item.index() - 2);
-      },
-      update: (evt, { item }) => {
-        const prevIndex = +$(this).attr('data-previndex');
-        length = tocEntries.children.length;
-        const newIndex = length - item.index() - 1;
-        const newOrder = Array.from(tocEntries.children)
-          .map((el, idx) => idx);
-        newOrder.splice(prevIndex, 1);
-        newOrder.splice(newIndex, 0, prevIndex);
-        this.props.updateLayerOrder(newOrder);
-      }
-    });
-  }
+    // Layer list is displayed in reverse order so index needs to reversed to get back to original reference.
+    const reverseIndex = (index) => {
+      return this.props.layerList.length - index - 1;
+    };
+
+    const prevIndex = reverseIndex(source.index);
+    const newIndex = reverseIndex(destination.index);
+    const newOrder = [];
+    for(let i = 0; i < this.props.layerList.length; i++) {
+      newOrder.push(i);
+    }
+    newOrder.splice(prevIndex, 1);
+    newOrder.splice(newIndex, 0, prevIndex);
+    this.props.updateLayerOrder(newOrder);
+  };
 
   _renderLayers() {
-    return this.props.layerList
-      .map((layer) => {
-        return (
-          <TOCEntry
-            key={layer.getId()}
-            layer={layer}
-          />
-        );
-      })
-      .reverse();
+    // Reverse layer list so first layer drawn on map is at the bottom and
+    // last layer drawn on map is at the top.
+    const reverseLayerList = [...this.props.layerList].reverse();
+
+    if (this.props.isReadOnly) {
+      return reverseLayerList
+        .map((layer) => {
+          return (
+            <TOCEntry
+              key={layer.getId()}
+              layer={layer}
+            />
+          );
+        });
+    }
+
+    return (
+      <EuiDragDropContext onDragEnd={this._onDragEnd}>
+        <EuiDroppable droppableId="layerTOC" spacing="none">
```

Comment:
You may want to be more specific with the droppableID just to ensure no conflicts